umount: disallow -r option for non-root users
(backport from 2.12r-pre1, fixes CAN-2005-2876)

diff -Nur util-linux-2.12q/mount/umount.c util-linux-2.12r/mount/umount.c
--- util-linux-2.12q/mount/umount.c	2004-12-20 23:03:45.000000000 +0100
+++ util-linux-2.12r/mount/umount.c	2005-09-10 20:07:38.000000000 +0200
@@ -714,7 +714,7 @@
 
 	if (getuid () != geteuid ()) {
 		suid = 1;
-		if (all || types || nomtab || force)
+		if (all || types || nomtab || force || remount)
 			die (2, _("umount: only root can do that"));
 	}
 
